tools/generate_crosstool/main.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	crosstoolpb "github.com/bazelbuild/bazel/src/main/protobuf/crosstool_config_go_proto"
+	"github.com/golang/protobuf/proto"
+)
+
+var (
+	out         = flag.String("out", "", "filename for CROSSTOOL text proto to write")
```

Comment:
Writing the proto to stdout seems easier and more flexible here

Reply:
you can always do `-out=/dev/stdout` if you want it on stdout. 

it felt more natural to write out to a file directly rather than rely on shell redirection.